redhat-appstudio/service-provider-integration-operator
# Allow file content requests from GitLab servers

### What does this PR do?

Allows to create & deliver content for FileContentRequests for GitLab repositories.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-253
https://issues.redhat.com/browse/SVPI-242

### How to test this PR?

 Deploy & create File request CR as follows:
 
 ```
 apiVersion: appstudio.redhat.com/v1beta1
kind: SPIFileContentRequest
metadata:
  name: test-file-content-request
  namespace: default
spec:
  repoUrl: https://gitlab.com/mshaposhnik/proj1
  filePath: foo/bar.txt
```

provide token either via OAuth or upload. Verify file content is injected.
